controllers/validator.go

```diff
@@ -100,6 +103,30 @@ func (r *DPAReconciler) ValidateVeleroPlugins(log logr.Logger) (bool, error) {
 				return false, err
 			}
 		}
+
+		// check csi compatibility with cluster version
+		// velero version <1.9 expects API group snapshot.storage.k8s.io/v1beta1, while OCP 4.11 (k8s 1.24) has only snapshot.storage.k8s.io/v1
+		for _, plugin := range dpa.Spec.Configuration.Velero.DefaultPlugins {
+			if plugin == oadpv1alpha1.DefaultPluginCSI {
+				clusterVersion, err := getClusterVersion()
+				if err != nil {
+					return false, err
+				}
+
+				if clusterVersion.Major == "1" && clusterVersion.Minor >= "24" {
```

Comment:
looks good! thank you

Reply:
Why are we doing nested for loops for no reason? The outerloop is already for each plugin. I would put this logic in line 90.
https://github.com/openshift/oadp-operator/blob/65159dd313799cc3ca52a0a4b31623098ee62900/controllers/validator.go#L89-L120

Reply:
This works until k8s 2.0, but I can only assume that oadp 1.0 won't be supported at all on whatever version of OpenShift that would be.

Reply:
@kaovilai @sseago earlier I had a function for this and then relocated the logic, copy/paste oversight, will rectify, Thank you !

Reply:
Done, updated the PR !